assets/js/frontend/course-video/video-blocks-extension.js

```diff
@@ -139,6 +139,46 @@
 		)
 		.forEach( initVimeoPlayer );
 
+	const initVideoPressPlayer = ( iframe ) => {
+		iframe.addEventListener( 'load', () => {
+			// eslint-disable-next-line @wordpress/no-global-event-listener
+			window.addEventListener(
+				'message',
+				( event ) => {
+					if ( event.source !== iframe.contentWindow ) {
+						return;
+					}
+					if ( event.data.event === 'ended' ) {
```

Comment:
I was just thinking if there's some other information in the data to check if it's a videopress video. Just because other things could post a message with `ended` in an event property. It would probably be a big coincidence, but it could happen. :P

If there's no another prop to check it, that's okay.

Reply:
I'm sure our good friends on Greenhouse would be happy to take a quick peek. 🙂

@Automattic/greenhouse For context, we're extending the VideoPress block to add support for pausing the video when the student navigates away, and to disable lesson completion until the video is watched. Would appreciate a review here and any tips you have to offer. 🥂

Reply:
👋 hey there

The code for the post message support is here, and we do pass along an `id` that is the video's guid. So you could verify that it is for the video you are expecting a message for

https://github.com/Automattic/videopress/blob/trunk/src/plugins/post-message-api/index.js#L43

Reply:
Hi @jgcaruso 👋 Thank you!

Could you please also help me with extracting the `id` of the video from the iframe's `src` attribute?
In my tests, I get URLs like: `https://video.wordpress.com/embed/{video-id}?cover=1&preloadContent=metadata&hd=1`. So I suppose that the last part of a path in the URL is an ID of the video. Is that a correct and reliable rule?

Reply:
> So I suppose that the last part of a path in the URL is an ID of the video. Is that a correct and reliable rule?

Yes! That is how we determine which video to load. Just fyi you may also see urls that look like this, depending on how users are embedding the video (with or without query string arguments):

video.wordpress.com/v/{video-id}
videopress.com/embed/{video-id}
videopress.com/v/{video-id}